src/playwright/playwright-plugin.ts

```diff
@@ -24,22 +26,48 @@ export class PlaywrightPlugin extends BrowserPlugin<BrowserType, Parameters<Brow
             ...launchOptions!.proxy,
         };
 
-        if (useIncognitoPages) {
-            browser = await this.library.launch(launchOptions);
-        } else {
-            const browserContext = await this.library.launchPersistentContext(userDataDir, launchOptions);
+        const [anonymizedProxyUrl, close] = await anonymizeProxySugar(proxyUrl);
+        if (anonymizedProxyUrl) {
+            launchOptions!.proxy = {
+                server: anonymizedProxyUrl,
+                bypass: launchOptions!.proxy.bypass,
+            };
+        }
+
+        try {
+            if (useIncognitoPages) {
+                browser = await this.library.launch(launchOptions);
+
+                if (anonymizedProxyUrl) {
+                    browser.on('disconnected', async () => {
+                        await close();
+                    });
+                }
+            } else {
+                const browserContext = await this.library.launchPersistentContext(userDataDir, launchOptions);
+
+                if (anonymizedProxyUrl) {
+                    browserContext.on('close', async () => {
+                        await close();
+                    });
+                }
+
+                if (!this._browserVersion) {
+                    // Launches unused browser just to get the browser version.
+                    const inactiveBrowser = await this.library.launch(launchOptions);
+                    this._browserVersion = inactiveBrowser.version();
 
-            if (!this._browserVersion) {
-                // Launches unused browser just to get the browser version.
-                const inactiveBrowser = await this.library.launch(launchOptions);
-                this._browserVersion = inactiveBrowser.version();
+                    inactiveBrowser.close().catch((error) => {
```

Comment:
not `await`ing here is on purpose? if so, what is the reason?

Reply:
I'm +1 for awaiting here but Ondra had said that we should skip this so I followed.

Reply:
We don't want to slow down just to make the `version` work. Nothing more really.

Reply:
Alternatively we could use CDP for this https://chromedevtools.github.io/devtools-protocol/tot/Browser/#method-getVersion but IIRC we cannot connect to CDP with persistent context, that's a playwright limitation.

https://github.com/microsoft/playwright/issues/13111